pubnub-gson/src/test/kotlin/com/pubnub/internal/v2/subscription/SubscriptionImplTest.kt

```diff
@@ -0,0 +1,77 @@
+package com.pubnub.internal.v2.subscription
+
+import com.pubnub.api.callbacks.Listener
+import com.pubnub.api.v2.callbacks.EventListener
+import com.pubnub.api.v2.subscriptions.EmptyOptions
+import com.pubnub.api.v2.subscriptions.SubscriptionOptions
+import com.pubnub.internal.PubNubCore
+import com.pubnub.internal.PubNubImpl
+import com.pubnub.internal.v2.callbacks.EventEmitterImpl
+import com.pubnub.internal.v2.callbacks.EventListenerCore
+import com.pubnub.internal.v2.entities.ChannelGroupName
+import com.pubnub.internal.v2.entities.ChannelName
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class SubscriptionImplTest {
+    private lateinit var objectUnderTest: TestableSubscriptionImpl
+
+    private val pubnub: PubNubImpl = mockk(relaxed = true)
+    private val pubnubCore: PubNubCore = mockk()
+    private val channels: Set<ChannelName> = emptySet()
+    private val channelGroups: Set<ChannelGroupName> = emptySet()
+    private val options: SubscriptionOptions = EmptyOptions
+    private val eventEmitter: EventEmitterImpl = mockk(relaxed = true)
+    private val listener: EventListenerCore = mockk()
+
+    @BeforeEach
+    internal fun setUp() {
+        every { pubnub.pubNubCore } returns pubnubCore
+        objectUnderTest = TestableSubscriptionImpl(pubnub, channels, channelGroups, options) { eventEmitter }
+    }
+
+    @Test
+    fun `addListener should call addListener on eventEmitter`() {
```

Comment:
this method isn't testing SubscriptionImpl, it's testing `BaseSubscriptionImpl.addListener(EventListenerCore)`


if you wanted to test `SubscriptionImpl.addListener(EventListener)`, you could check if it calls `addListener(EventListenerCore)` and passes in a `DelegatingEventListener` that is equals to `DelegatingEventListener(listener)` where listener is an `EventListener`